### ISSUE TITLE
Whatsie not showing up after opening some application in full screen

### ISSUE THREAD

**Issue by Kurobyte** [Aluxian/Whatsie#127]
I've recently opened a game at full screen, but after quitting the game, Whatsite doesn't show up, it still open but doesn't show up. And for fixing i've had to remove all cache & config data.

After some research I've found why is ocurring. In the `prefs.json` file at `windows-bounds` key has some values out of screen:
```json
"window-bounds":{"x":-32000,"y":-32000,"width":839,"height":650}
```

After changin that values manualy the app show up perfectly after a restart.

¿Could be possible add a context menu to the tray icon, to reset the windows-bounds position?

* Operating system: Win 7
* Whatsie version: 2.0.14-beta

**Comment by Aluxian** [comment]
Ugh. Oh. I'll add a check, if it goes off-screen to reset the position.

--
Alexandru Rosianu Full-Stack Developer aluxian.com | startersquad.com On Wed, Apr 27, 2016 at 5:33 PM, Kurobyte < notifications@github.com [notifications@github.com] > wrote:
I've recently opened a game at full screen, but after quitting the game, Whatsite doesn't show up, it still open but doesn't show up. And for fixing i've had to remove all cache & config data.

After some research I've found why is ocurring. In the prefs.json file at windows-bounds key has some values out of screen:

"window-bounds":{"x":-32000,"y":-32000,"width":839,"height":650}

After changin that values manualy the app show up perfectly after a restart.

¿Could be possible add a context menu to the tray icon, to reset the windows-bounds position?

 * Operating system: Win 7
 * Whatsie version: 2.0.14-beta

**Comment by Kurobyte** [comment]
Fyi: I'm using two screens, and if I place the window in the second screen
the 'X' value is negative.
Keep that in mind. If you want to prevent the XY value become less than 0.

2016-04-27 16:40 GMT+02:00 Alexandru Rosianu <notifications@github.com>:

> Ugh. Oh. I'll add a check, if it goes off-screen to reset the position.
>
> --
> Alexandru Rosianu Full-Stack Developer aluxian.com | startersquad.com On
> Wed, Apr 27, 2016 at 5:33 PM, Kurobyte < notifications@github.com [

**Comment by Aluxian** [comment]
Thanks for the heads up, I'll give it a second thought.


--
Alexandru Rosianu Full-Stack Developer aluxian.com | startersquad.com On Wed, Apr 27, 2016 at 6:47 PM, Kurobyte < notifications@github.com [notifications@github.com] > wrote:
Fyi: I'm using two screens, and if I place the window in the second screen
the 'X' value is negative.
Keep that in mind. If you want to prevent the XY value become less than 0.

2016-04-27 16:40 GMT+02:00 Alexandru Rosianu <notifications@github.com>:

> Ugh. Oh. I'll add a check, if it goes off-screen to reset the position.
>
> --
> Alexandru Rosianu Full-Stack Developer aluxian.com | startersquad.com On
> Wed, Apr 27, 2016 at 5:33 PM, Kurobyte < notifications@github.com [

**Comment by Aluxian** [comment]
I added the following checks:
- when the app is launched and the bounds are at -32000, the bounds are reset
- when the app window is focused and the bounds are at -32000, the bounds are reset

Also, I added an item in the tray menu to reset the window. I hope these are enough.

I was also thinking I could ask the user whether the window should be reset when Whatsie detects it is off-screen (width or height is negative, `x+width < 0` or `y+height < 0`), but this would be too complicated (I'd have to decide when to ask and which positions are actually on-screen screen and which are off). 

Probably this would be best handled by Electron, which could determine exactly for each platform when the window is really off-screen.

Atom currently does the checks I mentioned above (bounds are invalid if: width or height is negative or `x+width < 0` or `y+height < 0`), but I hesitated to do the same because you said position can be negative.
